node-js-basics-101/node_modules/asynquence-contrib/contrib.src.js

```diff
@@ -0,0 +1,2360 @@
+							};
+							// starting out with initial channel messages?
+							if (token.channel.messages.length > 0) {
+								// fake back-pressure blocking for each
+								token.channel.put_queue = token.channel.messages.map(function $$map() {
+									// make a notifiable iterable for 'put' blocking
+									return ASQ.iterable().then(function $$then() {
+										unblock(token.channel.take_queue.shift());
+										return !token.channel.closed;
+									});
+								});
+							}
+						}
+
+						// initialize the generator
+						it = gen.apply(ø, [token.channel].concat(args));
+
+						(function iterate() {
+
+							function next() {
+								// keep going with next step in goroutine?
+								if (!done) {
+									iterate();
+								}
+								// unblock overall goroutine handling to
+								// continue with other goroutines
+								else {
+										unblock();
+									}
+							}
+
+							// has a resumption value been achieved yet?
+							if (!ret) {
+								// try to resume the goroutine
+								try {
+									// resume with injected exception?
+									if (err) {
+										ret = it["throw"](err);
+										err = null;
+									}
+									// resume normally
+									else {
+											ret = it.next(msg);
+										}
+								}
+								// resumption failed, so bail
+								catch (e) {
+									done = true;
+									err = e;
+									msg = null;
+									unblock();
+									return;
+								}
+
+								// keep track of the result of the resumption
+								done = ret.done;
+								ret = ret.value;
+								type = typeof ret;
+
+								// if this goroutine is complete, unblock the
+								// overall goroutine handling
+								if (done) {
+									unblock();
+								}
+
+								// received a thenable/promise back?
+								if (isPromise(ret)) {
+									ret = ASQ().promise(ret);
+								}
+
+								// wait for the value?
+								if (ASQ.isSequence(ret)) {
+									ret.val(function $$val() {
+										ret = null;
+										msg = arguments.length > 1 ? ASQ.messages.apply(ø, arguments) : arguments[0];
+										next();
+									}).or(function $$or() {
+										ret = null;
+										msg = arguments.length > 1 ? ASQ.messages.apply(ø, arguments) : arguments[0];
+										if (msg instanceof Error) {
+											err = msg;
+											msg = null;
+										}
+										next();
+									});
+								}
+								// immediate value, prepare it to go right back in
+								else {
+										msg = ret;
+										ret = null;
+										next();
+									}
+							}
+						})();
+
+						// keep this goroutine alive until completion
+
+					case 7:
+						if (done) {
+							context$3$0.next = 17;
+							break;
+						}
+
+						context$3$0.next = 10;
+						return token;
+
+					case 10:
+						if (!(!done && !token.block && token.unblock_count === 0)) {
+							context$3$0.next = 14;
+							break;
+						}
+
+						context$3$0.next = 13;
+						return token.block = ASQ.iterable();
```

Comment:
__[jshint]__
* Did you mean to return a conditional instead of an assignment?

<!-- Comment by [SideCI](https://sideci.com) -->